Project.toml

```diff
@@ -41,6 +43,7 @@ ArgParse = "1"
 ArtifactWrappers = "0.2"
 Artifacts = "1"
 AtmosphericProfilesLibrary = "0.1.7"
+BlockArrays = "1"
```

Comment:
Can we make the new dependencies `weakdeps` and implement them as extensions?

Reply:
The BlockArrays dependency will be removed once some things are moved to ClimaCore. The ForwardDiff dependency cannot be an extension; it is required for autodiff, which we will be using for EDMF simulations for the foreseeable future.

Reply:
Being used for the foreseeable future and being and extensions are not conflicting. 

You can also make it an extension that is automatically loaded all the time. This will give us a path to make it a normal extension in the future

Reply:
I am extremely confused by what you are trying to achieve here. The point of this PR is to add two flags to our configuration: `use_exact_jacobian` and `debug_approximate_jacobian`. Setting either of these flags to `true` means that `ForwardDiff.jl` needs to be loaded. These flags will be used extensively in the development of implicit EDMF (and all other aspects of the implicit solver) moving forward.

As far as I am aware, the optional loading of packages based on YAML configurations is not the point of using extensions. Moreover, `ForwardDiff.jl` is already a dependency of `ClimaCore.jl`, `CloudMicrophysics.jl`, and `RootSolvers.jl`, so making it an extension will have no impact whatsoever on compilation time. So, I do not see any justification for making this an extension.

Reply:
What I am trying to achieve is isolating dependencies and providing pathways to making them optional or removing them. As you think of a target ClimaAtmos, is this feature a core feature, or is this an optional one? If it not a core feature, the target ClimaAtmos should not have a hard-dependency on `ForwardDiff`.

The fact that `ForwardDiff` is already used by three of our packages does not mean that we should not try to keep our footprint as small as reasonably possible or that we won't remove the dependency in the future from those three packages. 

In this case, turning it into an extension is trivial: `ForwardDiff` is only used in two functions, so you can just move the two functions into a file in `ext/` and put functions stubs in their place.

That said, I agree that `ForwardDiff` is currently used elsewhere (and this is unlikely to change any time soon) and turning into an extension might too much at this point. That said, I think we should have a mindset where optional functionalities should add no costs to packages. In this case, the cost is an extra direct dependency. I would like us to be mindful of this because never thinking about this is how we got environments with 500 packages that take 30 minutes to instantiate.

Reply:
Sorry if I wasn't clear earlier. Moving forward, this will be a core feature of ClimaAtmos. Similarly, the ability to solve a nonlinear equation using autodiff is a core feature of RootSolvers. I think this means both packages will continue to have ForwardDiff as a dependency.